Manage bloom indexed keys by procedure

ragadeeshu opened the pull request.

ragadeeshu committed:
Management of bloom indexed keys in procedure

ragadeeshu committed:
Update the BloomFulltextIndexBenchmarks

Also some minor cleanup and an additional test for the new procedure for
setting property keys.

chrisvest commented on enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextProvider.java:
```diff
@@ -244,6 +234,29 @@ public InternalIndexState getState( String identifier, FulltextIndexType type )
+        if ( type == FulltextIndexType.NODES )
+        {
+            nodeIndices.remove( identifier ).drop();
+            writableNodeIndices.remove( identifier ).close();
```
We need to protect these hashmaps from concurrent access.

chrisvest reviewed (changes requested):
Looks fine, except we need to protect the maps in the provider, now.

ragadeeshu committed:
Synchronize fulltextprovider register and drop

Also some minor cleanups

ragadeeshu commented:
@chrisvest updated

ragadeeshu committed:
Concurrent collections for the fulltextprovider

ragadeeshu committed:
Lock fulltextprovider while making schema changes

Dropping and registering indices takes a writelock, while the
transactioneventhandler takes a readlock in order to make sure that the
data used in aftercommit matches the data collected in beforecommit.

ragadeeshu committed:
Let update queue drain when dropping fulltextindex

This is so that no lucene writers holds locks on files that we need to
delete.

This commit also contains cleanup after a rebase
